ui/components/component-library/form-text-field/form-text-field.test.js

```diff
@@ -0,0 +1,365 @@
+/* eslint-disable jest/require-top-level-describe */
+import React, { useState } from 'react';
+import { fireEvent, getByLabelText, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { SIZES } from '../../../helpers/constants/design-system';
+
+import { FormTextField } from './form-text-field';
+
+// userEvent setup function as per testing-library docs
+// https://testing-library.com/docs/user-event/intr
+function setup(jsx) {
+  return {
+    user: userEvent.setup(),
+    ...render(jsx),
+  };
+}
+
+// Custom userEvent setup function that renders the component in a controlled environment.
+// This is used for the showClearButton and related props as the clearButton will only show in a controlled environment.
+function setupControlled(FormComponent, props) {
+  const ControlledWrapper = () => {
+    const [value, setValue] = useState('');
+    return (
+      <FormComponent
+        value={value}
+        onChange={(e) => setValue(e.target.value)}
+        {...props}
+      />
+    );
+  };
+  return { user: userEvent.setup(), ...render(<ControlledWrapper />) };
+}
```

Comment:
These should be moved to `test/lib/render-helpers.js` now that they are being used more than once.